dicetable.py

```diff
@@ -0,0 +1,327 @@
+
+
+class dicetable(object):
+    def __init__(self,dsize):
+        self.totaldice = 0
+        self.table = {0:1}
+        self.dsize = dsize
+        
+
+    def getTableVal(self,value):
+        return self.table.get(value,0)
+    def getTable(self):
```

Comment:
You've done a nice job of encapsulating the table object, this breaks that encapsulation. You do not want to expose this to clients of your API. They can do bad things with it.